docs/src/DeveloperDocumentation/styleguide.md

```diff
@@ -57,7 +57,13 @@ Here is a summary of the naming convention followed in OSCAR:
   also being consistent.
   For compatibility with standard Julia, while staying consistent internally,
   we also provide aliases (using `AbstractAlgebra.@alias`) for various standard
-  Julia functions, e.g. `is_one` as alias for `isone`
+  Julia functions, e.g. `is_one` as alias for `isone`.
+- A function returning the number of some things should be named `number_of_things`,
+  alternatively it can be named `n_things` with an alias to `number_of_things`.
+  The preferred style should be consistent throughout the corresponding part.
+  For some very common things, like the number of generators, we additionally
+  provide a shorter alias, e.g. `ngens` for `number_of_generators`. These aliases should be
+  very short and without underscores.
```

Comment:
```suggestion
  very short and without underscores, and one should only use
  them inside of functions, not in the user interface and documentation.
```
I think this is important to state: For consistency, I think we should attach docstrings not to `ngens` but to one of the other two, and the docstring should contain the signature of the "long form". It may additionally state the existence of the `ngens` alias, but that should not be too dominantly to not confuse users with three names for a single function.

Reply:
Since the changes are on a branch of a private repo (which I find unnecessarily tedious), I just write comments.

* `number_of` or `n_` but only one version per section. Example: `n_` is used throughout the polyhedral geometry section.
* `number_of` should be available as alias if the function is written as `n_`
* `n_` may be used as an alias if the function is written as `number_of`
* `ngens` should become `n_gens` then

Reply:
So it seems we agree at least in so far as: there should always be a `number_of_BLA` and there *may* be also a `n_BLA` (ignoring the technical details about e.g. which one may or may not be an "alias" (as defined via `@alias`) of which one for the moment).


But note that renaming `ngens` to `n_gens` would not quite fit: since we have `number_of_generators` it would have to be called `n_generators`... Moreover we use `ngens` and `gens` in ~1000 spots in Oscar.jl alone, and as Benjamin pointed out, it is used 20 times in book.

This is why in January a dozen of us agreed after multiple discussions over two weeks to allow certain "ultra short" aliases being grandfathered in... besides `ngens` that's also `nrows` and `ncols` and probably more (e.g. perhaps `nv` or `ne` for Graphs.jl compatibility?! (I am not a fan of those names, but if someone thinks this is useful to have, I don't want to block them?).


That said, I have no personal attachment to `ngens` and could easily adapt to using `n_gens` or even `n_generators`, but my point here is that this is a big change requiring a lot of work (with changes over multiple packages, once again carefully coordinated); the `n_gens` variant still is not really consistent; and in the end we can't get rid of `ngens` for now anyway. So is it really worth the trouble at this point?